main.go

```diff
@@ -61,26 +61,28 @@ func udpServer(address string, h *hub) {
 	for {
 		n, addr, err := serverConn.ReadFromUDP(buf)
 		if err != nil {
-			fmt.Println("Error: UDP read: %v", err)
+            log.WithFields(log.Fields{
+                "error": err,
+            }).Fatal("UDP read error")
 		}
 		if strings.HasPrefix(string(buf[:n]), "CONNECT") {
 			split := strings.Split(string(buf[:n]), ":")
 			privateKey := split[1]
 			name := split[2]
-			log.Println("New client:", privateKey+":"+name)
-			h.sendChannel <- sendPacket{addr: addr, data: []byte("HANDSHAKE")}
+            log.WithFields(log.Fields{
+                "privateKey": privateKey,
+                "name": name,
+            }).Info("New client")
+			h.sendChannel <- SendPacket{addr: addr, data: []byte("HANDSHAKE")}
 			if _, ok := h.clientMap[privateKey]; !ok {
-				h.clientMap[privateKey] = make(map[string]*client)
+				h.clientMap[privateKey] = make(map[string]*Client)
 			}
 			if _, ok := h.clientMap[privateKey][name]; !ok {
-				cli := &client{addr: addr, process: make(chan []byte, 1),
+				cli := &Client{addr: addr, process: make(chan []byte, 1),
 					name: name, privateKey: privateKey, h: h,
 					rateLoggers: make(map[string]*loggers.RateLogger)}
 				h.clientMap[privateKey][name] = cli
 				addrMap[string(addr.IP)+":"+string(addr.Port)] = cli
-				if db != dbNone {
-					h.dbw.SetAdd(true, "clients:list", name)
-				}
 				go cli.processor()
 			} else {
 				h.clientMap[privateKey][name].addr = addr
```

Comment:
On Line 89 and Line 92 (I can't comment on the actual lines for some reason), might be cleaner to move the `IP:Port` string concatenation into a small helper function, instead of doing it all inline. Or at least create the string separately in a variable instead of doing it inside `addrMap[...]`.